spec/smart_app_launch/well_known_capabilities_test_spec.rb

```diff
@@ -0,0 +1,144 @@
+        expect(result.result_message).to match(/must be type: string/)
+      end
+
+      config = valid_config.dup
+      config['capabilities'] = '1'
+      result = run(runnable, well_known_configuration: config.to_json)
+
+      expect(result.result).to eq('fail')
+      expect(result.result_message).to match(/must be type: array/)
+    end
+
+    it 'fails if the capabilities field contains a non-string entry' do
+      config = valid_config.dup
+      config['capabilities'] << 1
+      config['capabilities'] << nil
+      result = run(runnable, well_known_configuration: config.to_json)
+
+      expect(result.result).to eq('fail')
+      expect(result.result_message).to match(/must be an array of strings/)
+      expect(result.result_message).to match(/`1`/)
+      expect(result.result_message).to match(/`nil`/)
+    end
+  end
+
+  describe SMARTAppLaunch::WellKnownCapabilitiesSTU1Test do
+    it_behaves_like 'well-known tests' do
+      let(:runnable) { test_v1 }
+      let(:required_fields) { ['authorization_endpoint', 'token_endpoint', 'capabilities'] }
+      let(:valid_config) { well_known_config.slice(*required_fields) }
+    end
+  end
+
+  describe SMARTAppLaunch::WellKnownCapabilitiesSTU2Test do
+    let(:required_fields) { ['authorization_endpoint', 'token_endpoint', 'capabilities', 'issuer', 'jwks_uri', 'grant_types_supported', 'code_challenge_methods_supported'] }
+    let(:runnable) { test_v2 }
+    let(:valid_config) { well_known_config.slice(*required_fields) }
+
+    it_behaves_like 'well-known tests'
+
+    it 'fails if `issuer` is missing while `sso-openid-connect` is listed as a capability' do
+      config = valid_config.dup
+      config.delete('issuer')
+      result = run(runnable, well_known_configuration: config.to_json)
+      expect(result.result).to eq('fail')
+      expect(result.result_message).to match('Well-known `issuer` field must be a string and present when server capabilities includes `sso-openid-connect`')
+    end
+
+    it 'fails if `jwks_uri` is missing while `sso-openid-connect` is listed as a capability' do
+      config = valid_config.dup
+      config.delete('jwks_uri')
+      result = run(runnable, well_known_configuration: config.to_json)
+      expect(result.result).to eq('fail')
+      expect(result.result_message).to match('Well-known `jwks_uri` field must be a string and present when server capabilites includes `sso-openid-coneect`')
+    end
+
+    it 'warns if `issuer` is present while `sso-openid-connect` is not listed as a capability' do
+      config = valid_config.dup
+      config['capabilities'].reject! { |capability| capability == 'sso-openid-connect' }
+      result = run(runnable, well_known_configuration: config.to_json)
+      expect(result.result).to eq('pass')
+      warning_messages = Inferno::Repositories::Messages.new.messages_for_result(result.id).filter { |message| message.type == 'warning' }
+      expect(warning_messages).to be_present
```

Comment:
```suggestion
```
`filter` will always return an array, so if `warning_messages` == empty array, then `any?` will return `false` and the second expect will fail. so, no need to explicitly check